lib/phil/charlie_cards/charlie_card.ex

```diff
@@ -0,0 +1,48 @@
+defmodule Phil.CharlieCards.CharlieCard do
+  @moduledoc """
+  A physical piece of fare media.
+
+  Batch number - Indicates in which shipment from the manufacturer this card arrived.
+  Batch sequence number - The sequential number of this card within its batch.
+  Card valid from - Date the card will physically start working.
+  Card valid until - Date the card will physically stop working.
+  Product - The type of card: Youth Pass, Senior Pass, etc.
+  Product valid from - Date the card can start being used.
+  Product valid until - Date after which the card can no longer be used.
+  Production date - Date the card was manufactured.
+  Sequence number - The overall sequence number of the card across all batches.
+  Serial number - A unique identifier for the card.
+  Status - Status of the card: available or unknown, currently.
+  """
+
+  use Ecto.Schema
+
+  @products ~w(youth_pass)a
+  @statuses ~w(available unknown)a
+
+  schema "charlie_cards" do
+    field(:batch_number, :integer)
+    field(:batch_sequence_number, :integer)
+    field(:card_valid_from, :utc_datetime)
+    field(:card_valid_until, :utc_datetime)
+    field(:product, Ecto.Enum, values: @products)
+    field(:product_valid_from, :utc_datetime)
+    field(:product_valid_until, :utc_datetime)
+    field(:production_date, :utc_datetime)
+    field(:sequence_number, :integer)
+    field(:serial_number, :integer)
```

Comment:
is this always an integer? Or is there some chance of errant letters, I'd be tempted to make it a string given that we have sequence numbers as well...

Reply:
I'm fairly certain that it is always an integer on their side. But, I don't have any problems with making it a string just in case.

Reply:
these are always integers

Reply:
Upload doesn't have `5-` in the serial number. There is a mediatype field that is apparently always `5` for MBTA. Probably where the `5-` comes from.